flake.nix

```
@@ -19,7 +19,7 @@
       pkgs = nixpkgs.legacyPackages.${system};
       corepackEnable = pkgs.runCommand "corepack-enable" {} ''
         mkdir -p $out/bin
-        ${pkgs.nodejs-18_x}/bin/corepack enable --install-directory $out/bin
+        ${pkgs.nodejs}/bin/corepack enable --install-directory $out/bin
```

We should always pin to a specific version of NodeJS with Nixpkgs.

```suggestion
        ${pkgs.nodejs_20}/bin/corepack enable --install-directory $out/bin
```

---

Hm. Did that, but can you clarify why that's necessary given that we have that locked in flake.lock? Protecting against accidentally upgrading past the major version?

---

Precisely - because Nixpkgs `nodejs` package is not pinned to a specific version of NodeJS, if we update our flake inputs (which we really should be doing regularly and there's CI to help us with this) we'd risk accidentally upgrading to an unintended version of NodeJS.

Also I had no idea there was even a `nodejs` (sans version) package in Nixpkgs.